Source/BlueprintSearchVSExtension/Source/Commands/CommandHelpers/PathFinderHelper.cs

```diff
@@ -27,82 +27,127 @@ public static class PathFinderHelper
 
 		private const char QuoteChar = '\"';
 
-
-		public static bool FindPaths()
+		public static bool FindUEProject(out string FoundProjectPath)
 		{
+			FoundProjectPath = string.Empty;
+
+			Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread();
+
 			List<Project> ProjectsList = new List<Project>();
 			ProjectsList.AddRange(DTEService.Solution.Projects.Cast<Project>());
```

Comment:
DTE compat is one of the things that might prove a headache when moving the plugin to 2022 (no action on this review just something you might want to be aware of)